### ISSUE TITLE
Dropdown responsive alignment

### ISSUE THREAD

**Issue by gauravshah27** [reactstrap/reactstrap#2159]
- components: `DropdownMenu`
- reactstrap version `9.0.0-0`
- import method `umd/csj/es` 
- react version `17.0.1`
- bootstrap version `5.0.0-beta3`

### What is happening?

Trying to implement Responsive Alignment for `DropdownMenu` using `dropdown-menu-*-start/end` and unable to get it to work

### What should be happening?

1. When using `dropdown-menu-end dropdown-menu-lg-start` the dropdown menu should open to the right by default but should shift to the left when breakpoint reaches lg or higher.

### Steps to reproduce issue

1. Update the DropdownMenu component in any example of reactstrap dropdown documentation and use 
`<DropdownMenu className='dropdown-menu-end dropdown-menu-lg-start'>` 
2. The Dropdown Menu should open on the left when operating on `lg` or higher breakpoints.
3. However, the menu keeps opening up on the right always.

**Comment by kyletsang** [comment]
I believe responsive alignment requires that popper be disabled, but I don't think that's currently possible.

**Comment by gauravshah27** [comment]
@kyletsang i am not sure if there is a change in behavior but even bootstrap 5 is using popper for placement of the menu and in the bootstrap 4.6 version, I was able to make responsive alignment work by replacing the 
`DropdownMenu` component with the html from bootstrap documentation (plus the onclick handler on button and conditionally adding `show` to toggle the menu visibility) and using the `dropdown-menu-lg-left/right` classes on the menu. I have tried using the same approach here with `dropdown-menu-start/end` but this time it does not work

**Comment by gauravshah27** [comment]
```javascript
import {useState} from 'react';
import {ButtonDropdown, DropdownToggle} from 'reactstrap';
import classnames from 'classnames';

export default function DemoDropdownResponsiveAlignment() {
  const [dropdownOpen1, setDropdownOpen1] = useState(false);
  const [dropdownOpen2, setDropdownOpen2] = useState(false);

  function toggleDropdown1() {
    setDropdownOpen1(!dropdownOpen1);
  }

  function toggleDropdown2() {
    setDropdownOpen2(!dropdownOpen2);
  }

  return (
    <>
      <ButtonDropdown isOpen={dropdownOpen1} toggle={toggleDropdown1}>
        <DropdownToggle caret>
          Left-aligned but right aligned when large screen
        </DropdownToggle>
        <div className={classnames('dropdown-menu', 'dropdown-menu-lg-right', {show: dropdownOpen1})}>
          <button className="dropdown-item" type="button">Action</button>
          <button className="dropdown-item" type="button">Another action</button>
          <button className="dropdown-item" type="button">Something else here</button>
        </div>
      </ButtonDropdown>
      <ButtonDropdown isOpen={dropdownOpen2} toggle={toggleDropdown2}>
        <DropdownToggle caret>
          Right-aligned but left aligned when large screen
        </DropdownToggle>
        <div
          className={classnames('dropdown-menu', 'dropdown-menu-right', 'dropdown-menu-lg-left', {show: dropdownOpen2})}
        >
          <button className="dropdown-item" type="button">Action</button>
          <button className="dropdown-item" type="button">Another action</button>
          <button className="dropdown-item" type="button">Something else here</button>
        </div>
      </ButtonDropdown>
    </>
  );
}
```

This code was working as expected for the responsive alignment for the following version numbers:
bootstrap: `4.6.0`
react: `17.0.1`
reactstrap: `8.9.0`

However if I replace `dropdown-menu-right` with `dropdown-menu-end` and `dropdown-menu-lg-left` with `dropdown-menu-lg-start` - I was expecting this to work (to your point there has been no change in the behavior for responsive alignment) but this does not work and I am trying to figure out what changed.

**Comment by gauravshah27** [comment]
@kyletsang never mind I found the culprit. `data-bs-popper` is the attribute that needs to be toggled when you are using the dropdown-menu. I inspected the code on the bootstrap site and found that attribute being toggled everytime I toggled  and toggling that in my react code gives the responsive alignment that I am looking for. Having said that - I do feel like there should be something in reactstrap that will allow us to use responsive alignment without having to use the hack that you mentioned. 

Thanks for all the information. I will be closing this issue now.
